control-plane-operator/controllers/hostedcontrolplane/cloud/powervs/powervs.go

```diff
@@ -0,0 +1,207 @@
+	"fmt"
+	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
+	"github.com/openshift/hypershift/support/releaseinfo"
+	"strings"
+	"text/template"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+const (
+	ccmContainerName       = "cloud-controller-manager"
+	kubeConfigFileBasePath = "/etc/kubernetes"
+	secretMountPath        = "/etc/vpc"
+	ccmConfigMapMountPath  = "/etc/ibm"
+)
+
+var ccmConfigTemplateData = `
+[global]
+version = 1.1.0
+[kubernetes]
+config-file = {{.ConfigFile}}
+[provider]
+cluster-default-provider = g2
+accountID = {{.AccountID}}
+clusterID = {{.ClusterID}}
+g2workerServiceAccountID = {{.G2workerServiceAccountID}}
+g2Credentials = {{.G2Credentials}}
+g2ResourceGroupName = {{.G2ResourceGroupName}}
+g2VpcSubnetNames = {{.G2VpcSubnetNames}}
+g2VpcName = {{.G2VpcName}}
+region = {{.Region}}
+powerVSCloudInstanceID = {{.PowerVSCloudInstanceID}}
+powerVSRegion = {{.PowerVSRegion}}
+powerVSZone = {{.PowerVSZone}}`
+
+var ccmConfigTemplate = template.Must(template.New("ccmConfigMap").Parse(ccmConfigTemplateData))
+
+func ReconcileCCMConfigMap(ccmConfig *corev1.ConfigMap, hcp *hyperv1.HostedControlPlane) (err error) {
+	config := map[string]string{
+		"ConfigFile":               fmt.Sprintf("%s/kubeconfig", kubeConfigFileBasePath),
+		"AccountID":                hcp.Spec.Platform.PowerVS.AccountID,
+		"ClusterID":                hcp.Name,
+		"G2workerServiceAccountID": hcp.Spec.Platform.PowerVS.AccountID,
+		"G2Credentials":            fmt.Sprintf("%s/%s", secretMountPath, "ibmcloud_api_key"),
+		"G2ResourceGroupName":      hcp.Spec.Platform.PowerVS.ResourceGroup,
+		"G2VpcSubnetNames":         hcp.Spec.Platform.PowerVS.VPC.Subnet,
+		"G2VpcName":                hcp.Spec.Platform.PowerVS.VPC.Name,
+		"Region":                   hcp.Spec.Platform.PowerVS.VPC.Region,
+		"PowerVSCloudInstanceID":   hcp.Spec.Platform.PowerVS.ServiceInstanceID,
+		"PowerVSRegion":            hcp.Spec.Platform.PowerVS.Region,
+		"PowerVSZone":              hcp.Spec.Platform.PowerVS.Zone,
+	}
+
+	configData := &bytes.Buffer{}
+	ccmConfigTemplate.Execute(configData, config)
+
+	if ccmConfig.Data == nil {
+		ccmConfig.Data = map[string]string{}
+	}
+
+	ccmConfig.Data[ccmConfig.Name] = configData.String()
+
+	return
+}
+
+func ReconcileCCMDeployment(deployment *appsv1.Deployment, hcp *hyperv1.HostedControlPlane, ccmConfig *corev1.ConfigMap, releaseImage *releaseinfo.ReleaseImage) (err error) {
+	commandToExec := []string{
+		"/bin/ibm-cloud-controller-manager",
+		"--authentication-skip-lookup",
+		"--bind-address=$(POD_IP_ADDRESS)",
+		"--use-service-account-credentials=true",
+		"--configure-cloud-routes=false",
+		"--cloud-provider=ibm",
+		fmt.Sprintf("--cloud-config=%s/%s", ccmConfigMapMountPath, ccmConfig.Name),
+		"--profiling=false",
+		"--leader-elect=true",
+		"--leader-elect-lease-duration=137s",
```

Comment:
does [this](https://github.com/openshift/hypershift/pull/1427/files#diff-e406d0d828faea9ebe5d40499c489a1e67c7872b278294581badc5e6c09afb98R22) apply here cc @alvaroaleman?

Reply:
This is inline with the openshift guidance to be able to tolerate approximately 1 minute of API disconnectivity, we saw no reason to deviate from this with CCMs because their operations are typically less time sensitive than the KCM operations